src/rgw/rgw_op.cc

```diff
@@ -4843,6 +4843,16 @@ void RGWPutLC::execute()
   RGWLCXMLParser_S3 parser(s->cct);
   RGWLifecycleConfiguration_S3 new_config(s->cct);
 
+  content_md5 = s->info.env->get("HTTP_CONTENT_MD5");
+  if (content_md5 == nullptr) {
+    op_ret = -ERR_INVALID_REQUEST;
+    s->err.message = "Missing required header for this request: Content-MD5";
+    ldout(s->cct, 5) << s->err.message << dendl;
+    return;
+  }
+
+  std::string content_md5_bin = rgw::from_base64(boost::string_view(content_md5));
```

Comment:
need to catch exceptions from `from_base64()`, which will throw if the input isn't base64

Reply:
@cbodley Done